### ISSUE TITLE
pynamodb error with boto3 1.8+ ('Endpoint' object has no attribute 'proxies')

### ISSUE THREAD

**Issue by sandroden** [pynamodb/PynamoDB#533]
While upgrading from boto3 1.7.84 to 1.8* a simple scan of a simple table raises an error. I'm using pynamodb 3.3: is it ready for 1.8+ series?

My simple table is define as:

```
class User(Model):

    class Meta:
        table_name = USERS_TABLE_NAME
        region = REGION

    user_id = attributes.UnicodeAttribute(hash_key=True)
    date_start = attributes.UTCDateTimeAttribute()
    date_last_play = attributes.UTCDateTimeAttribute(null=True)

    @classmethod
    def get_lazy_players(cls, date=None, days=None):
        date = (date or datetime.now()).astimezone(pytz.UTC)
        if days:
            date = date - timedelta(days=days)
        return cls.scan(cls.date_last_play <= date)
```

And a scan raises this error:

```
In [1]: from gamecontroller.dynamo_models import User

In [2]: User.scan()
Out[2]: <pynamodb.pagination.ResultIterator at 0x7fbad48b0d30>

In [3]: list(User.scan())
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-6c742096e1e4> in <module>()
----> 1 list(User.scan())

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/pagination.py in __next__(self)
    170         if self._first_iteration:
    171             self._first_iteration = False
--> 172             self._get_next_page()
    173 
    174         while self._index == self._count:

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/pagination.py in _get_next_page(self)
    155 
    156     def _get_next_page(self):
--> 157         page = next(self.page_iter)
    158         self._count = page[CAMEL_COUNT]
    159         self._items = page.get(ITEMS)  # not returned if 'Select' is set to 'COUNT'

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/pagination.py in __next__(self)
    100             self._rate_limiter.acquire()
    101             self._kwargs['return_consumed_capacity'] = TOTAL
--> 102         page = self._operation(*self._args, **self._kwargs)
    103         self._last_evaluated_key = page.get(LAST_EVALUATED_KEY)
    104         self._total_scanned_count += page[SCANNED_COUNT]

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/connection/table.py in scan(self, filter_condition, attributes_to_get, limit, conditional_operator, scan_filter, return_consumed_capacity, segment, total_segments, exclusive_start_key, consistent_read, index_name)
    219             exclusive_start_key=exclusive_start_key,
    220             consistent_read=consistent_read,
--> 221             index_name=index_name)
    222 
    223     def query(self,

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/connection/base.py in scan(self, table_name, filter_condition, attributes_to_get, limit, conditional_operator, scan_filter, return_consumed_capacity, exclusive_start_key, segment, total_segments, consistent_read, index_name)
   1290 
   1291         try:
-> 1292             return self.dispatch(SCAN, operation_kwargs)
   1293         except BOTOCORE_EXCEPTIONS as e:
   1294             raise ScanError("Failed to scan table: {0}".format(e), e)

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/connection/base.py in dispatch(self, operation_name, operation_kwargs)
    311 
    312         self.send_pre_boto_callback(operation_name, req_uuid, table_name)
--> 313         data = self._make_api_call(operation_name, operation_kwargs)
    314         self.send_post_boto_callback(operation_name, req_uuid, table_name)
    315 

~/.virtualenvs/mygenerali/lib/python3.6/site-packages/pynamodb/connection/base.py in _make_api_call(self, operation_name, operation_kwargs)
    355                     prepared_request,
    356                     timeout=self._request_timeout_seconds,
--> 357                     proxies=self.client._endpoint.proxies,
    358                 )
    359                 data = response.json()

AttributeError: 'Endpoint' object has no attribute 'proxies'

```

**Comment by bernard357** [comment]
The error message has been flagged in #529 

As explained there, a workaround is to set an explicit version of botocore in `requirements.txt`. The most recent version of botocore that works is the last 1.10 release, so I ended up with:

```
boto3
botocore==1.10.84
```

Hoping that this helps

**Comment by shadycuz** [comment]
Also seeing this while doing multiple things. Thanks.

**Comment by mikegrima** [comment]
I started seeing this too -- was pulling my hair out.

**Comment by garrettheel** [comment]
This should be fixed in [3.3.1](https://github.com/pynamodb/PynamoDB/releases/tag/3.3.1), thanks for reporting

**Comment by andreborges-janrain** [comment]
Hi,
I tried to update pynamodb to 3.3.1 but I still able to see this error.

**Comment by pilgrim2go** [comment]
I got the same issue - upgraded to 3.3.1 doesn't help

**Comment by sleon-fmi** [comment]
As of June 2019 I'm getting this error with:
- `pynamodb==3.3.0`
- `boto3==1.9.42`
- `botocore==1.12.42`

Is there a planned fix for this? Is `botocore==1.10.84` still the latest version known to not have this issue?

**Comment by ikonst** [comment]
@sleon-fmi it is fixed in pynamodb 3.3.1, not 3.3.0

**Comment by sleon-fmi** [comment]
> @sleon-fmi it is fixed in pynamodb 3.3.1, not 3.3.0

Will check it, thank you!
